geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/RepeatedRebalanceDUnitTest.java

```diff
@@ -0,0 +1,224 @@
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Region;
+import org.apache.geode.management.internal.cli.i18n.CliStrings;
+import org.apache.geode.test.dunit.rules.ClusterStartupRule;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.junit.assertions.TabularResultModelAssert;
+import org.apache.geode.test.junit.rules.GfshCommandRule;
+
+public class RepeatedRebalanceDUnitTest {
+  private static final String PARENT_REGION = "RepeatedRebalanceTestRegion";
+  private static final String COLOCATED_REGION_ONE = "RepeatedRebalanceColocatedRegionOne";
+  private static final String COLOCATED_REGION_TWO = "RepeatedRebalanceColocatedRegionTwo";
+  private static final String PARTITION_RESOLVER =
+      "org.apache.geode.management.internal.cli.commands.RepeatedRebalancePartitionResolver";
+
+  private static final int TOTAL_NUM_BUCKETS = 48;
+  private static final int NUM_REDUNDANT_COPIES = 2;
+  private static final int INITIAL_SERVERS = 4;
+  private static final int NUMBER_OF_ENTRIES = 30000;
+
+  private List<MemberVM> memberList = new ArrayList();
+  private MemberVM locator1;
+
+  @Rule
+  public GfshCommandRule gfsh = new GfshCommandRule();
+
+  @Rule
+  public ClusterStartupRule cluster = new ClusterStartupRule();
+
+
+  @Before
+  public void before() throws Exception {
+
+    locator1 = cluster.startLocatorVM(0);
+
+    for (int i = 0; i < INITIAL_SERVERS; ++i) {
+      memberList.add(cluster.startServerVM(i + 1, locator1.getPort()));
+    }
+
+    gfsh.connectAndVerify(locator1);
+
+    gfsh.executeAndAssertThat("create region --name=" + PARENT_REGION
+        + " --type=PARTITION --redundant-copies=" + NUM_REDUNDANT_COPIES
+        + " --enable-statistics=true "
+        + "--recovery-delay=-1 --startup-recovery-delay=-1 --total-num-buckets=" + TOTAL_NUM_BUCKETS
+        + " --partition-resolver=" + PARTITION_RESOLVER)
+        .statusIsSuccess();
+
+    gfsh.executeAndAssertThat("create region --name=" + COLOCATED_REGION_ONE
+        + " --type=PARTITION --redundant-copies=" + NUM_REDUNDANT_COPIES
+        + " --enable-statistics=true "
+        + "--recovery-delay=-1 --startup-recovery-delay=-1 --total-num-buckets=" + TOTAL_NUM_BUCKETS
+        + " --partition-resolver=" + PARTITION_RESOLVER
+        + " --colocated-with=" + PARENT_REGION)
+        .statusIsSuccess();
+
+    gfsh.executeAndAssertThat("create region --name=" + COLOCATED_REGION_TWO
+        + " --type=PARTITION --redundant-copies=" + NUM_REDUNDANT_COPIES
+        + " --enable-statistics=true "
+        + "--recovery-delay=-1 --startup-recovery-delay=-1 --total-num-buckets=" + TOTAL_NUM_BUCKETS
+        + " --partition-resolver=" + PARTITION_RESOLVER
+        + " --colocated-with=" + COLOCATED_REGION_ONE)
+        .statusIsSuccess();
+  }
+
+  @After
+  public void after() {
+    for (int i = 0; i < memberList.size(); i++) {
+      memberList.get(i).stop();
```

Comment:
you don't need to do this, the rule automatically stops all the member vms it started.

Reply:
Thanks, this has now been removed.